File: packages/image-labeling/android/src/main/java/io/capawesome/capacitorjs/plugins/mlkit/imagelabeling/classes/results/ProcessImageResult.java
package io.capawesome.capacitorjs.plugins.mlkit.imagelabeling.classes.results;

import androidx.annotation.NonNull;
import com.getcapacitor.JSArray;
import com.getcapacitor.JSObject;
import com.google.mlkit.vision.label.ImageLabel;
import io.capawesome.capacitorjs.plugins.mlkit.imagelabeling.interfaces.Result;
import java.util.List;

public class ProcessImageResult implements Result {

    @NonNull
    private final List<ImageLabel> labels;

    public ProcessImageResult(@NonNull List<ImageLabel> labels) {
        this.labels = labels;
    }

    @Override
    @NonNull
    public JSObject toJSObject() {
        JSArray labelsResult = this.createLabelsResult();

        JSObject result = new JSObject();
        result.put("labels", labelsResult);
        return result;
    }

    private JSArray createLabelsResult() {
        JSArray result = new JSArray();
        for (ImageLabel label : labels) {
            JSObject labelResult = this.createLabelResult(label);
            result.put(labelResult);
        }
        return result;
    }

    private JSObject createLabelResult(@NonNull ImageLabel label) {
        JSObject result = new JSObject();
        result.put("index", label.getIndex());
        result.put("text", label.getText());
        result.put("confidence", label.getConfidence());
        return result;
    }
}

File: packages/image-labeling/ios/Plugin/Classes/Results/ProcessImageResult.swift
import Foundation
import Capacitor
import MLKitImageLabeling

@objc public class ProcessImageResult: NSObject, Result {
    let labels: [ImageLabel]

    init(labels: [ImageLabel]) {
        self.labels = labels
    }

    @objc public func toJSObject() -> AnyObject {
        let labelsResult = createLabelsResult()

        var result = JSObject()
        result["labels"] = labelsResult

        return result as AnyObject
    }

    private func createLabelsResult() -> JSArray {
        var result = JSArray()

        for label in labels {
            let labelResult = createLabelResult(label)
            result.append(labelResult)
        }

        return result
    }

    private func createLabelResult(_ label: ImageLabel) -> JSObject {
        var result = JSObject()
        result["index"] = label.index
        result["text"] = label.text
        result["confidence"] = label.confidence
        return result
    }
}
